repository/git_testing.go

```diff
@@ -42,6 +53,95 @@ func CreateTestRepo(bare bool) *GitRepo {
 	return repo
 }
 
+// setupSigningKey creates a GPG key and sets up the local config so it's used.
+// The key id is set as "user.signingkey". For the key to be found, a `gpg`
+// wrapper which uses only a custom keyring is created and set as "gpg.program".
+// Finally "commit.gpgsign" is set to true so the signing takes place.
+func (repo *GitRepo) setupSigningKey() {
+	config := repo.LocalConfig()
+
+	// Generate a key pair for signing commits.
+	entity, err := openpgp.NewEntity("First Last", "", "fl@example.org", nil)
+	check(err)
+
+	if err := config.StoreString("user.signingkey", entity.PrivateKey.KeyIdString()); err != nil {
+		log.Fatal("failed to set user.signingkey for test repository: ", err)
+	}
+
+	// Armor the private part.
+	privBuilder := &strings.Builder{}
+	w, err := armor.Encode(privBuilder, openpgp.PrivateKeyType, nil)
+	check(err)
```

Comment:
Please use `assert.NoError(t, err)` or `require.NoError(t, err)` if you want to stop the test immediately.